include/ozo/ext/std/duration.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include <ozo/io/send.h>
+#include <ozo/io/recv.h>
+#include <ozo/type_traits.h>
+#include <boost/fusion/adapted/struct/define_struct.hpp>
+
+#include <chrono>
+#include <cstdint>
+
+BOOST_FUSION_DEFINE_STRUCT((ozo)(detail), pg_interval,
+    (std::int64_t, microseconds)
+    (std::int32_t, days)
+    (std::int32_t, months)
+)
+
+namespace ozo::detail {
+
+inline detail::pg_interval from_chrono_duration(const std::chrono::microseconds& in) {
+    static_assert(
+        std::chrono::microseconds::min().count() == std::numeric_limits<std::int64_t>::min() &&
+        std::chrono::microseconds::max().count() == std::numeric_limits<std::int64_t>::max(),
+        "std::chrono::microseconds tick representation type is not supported"
+    );
+
+    using days = std::chrono::duration<std::int32_t, std::ratio<24 * std::chrono::hours::period::num>>;
+
+    return detail::pg_interval{(in % days(1)).count(), std::chrono::duration_cast<days>(in).count(), 0};
+}
+
+inline std::chrono::microseconds to_chrono_duration(const pg_interval& interval) {
+    static_assert(
+        std::chrono::microseconds::min().count() == std::numeric_limits<std::int64_t>::min() &&
+        std::chrono::microseconds::max().count() == std::numeric_limits<std::int64_t>::max(),
+        "std::chrono::microseconds tick representation type is not supported"
+    );
+
+    using std::chrono::microseconds;
+    using std::chrono::duration_cast;
+
+    using usecs = std::chrono::duration<std::int64_t, std::micro>;
+    using days = std::chrono::duration<std::int64_t, std::ratio<24 * std::chrono::hours::period::num>>;
+    using months = std::chrono::duration<std::int32_t, std::ratio<30 * days::period::num>>;
+
+    auto usecs_surplus = usecs(interval.microseconds) % days(1);
+    auto days_total = months(interval.months) + days(interval.days) + duration_cast<days>(usecs(interval.microseconds));
+
+    if (days_total < (duration_cast<days>(microseconds::min()) - days(1))
+        || ((days_total < days(0)) && ((days_total + days(1)) + usecs_surplus < microseconds::min() + days(1)))
+        || (duration_cast<days>(microseconds::max()) + days(1)) < days_total
+        || ((days(0) < days_total) && (microseconds::max() - days(1) < (days_total - days(1)) + usecs_surplus))) {
+        throw std::overflow_error("duration overflow: can not store interval value in chrono::duration");
```

Comment:
I think it was a bad idea to throw an exception on overflow/underflow. I've just imaged how an unlucky client application started to crash because of somebody insert a row with an out of range value. Data bug with no chances to find it. So maybe it is better to set the variable into an appropriate limit value and to describe such behavior in the documentation?